authorization-api-1_0.md

```diff
@@ -597,9 +600,186 @@ X-Request-ID: bfe9eb29-ab87-4ca3-be83-a1d5d8305720
+
+pageToken:
+: OPTIONAL. A string value that is returned in a previous Subject Search Response ({{subject-search-response}}), which indicates that the request is a continuation of a previous request
+
+pageSize:
+: OPTIONAL. The maximum number of `decision` items in a Subject Search Response ({{subject-search-response}}). The API MAY return a smaller number of items but SHOULD NOT return a number of items that is greater than this value
+
+The content of a Subject Search Request body is a JSON object with the following fields:
+
+resource:
+: REQUIRED. A resource as described in the Resources section ({{resources}})
+
+queries:
+: REQUIRED. An array of `string` values as described in the Actions section ({{actions}}).
+
+The following is a non-normative example of a Subject Search Request:
+
+~~~ http
+POST /subjectsearch HTTP/1.1
+Host: pdp.mycompany.com?pageToken="NWU0OGFiZTItNjI1My00NTQ5LWEzYTctNWQ1YmE1MmVmM2Q4"&pageSize=2
+Authorization: <myoauthtoken>
+
+{
+  "resource": {
+    "id": "somevalue",
+    "type": "document",
+    "attributeNames": [
+        "author",
+        "createDate",
+        "lastUpdated"
+    ]
+   },
+  "queries": ["delete", "read", "write"]
+}
+~~~
+{: #example-subject-search-request title="Example Subject Search Request"}
+
+### Subject Search Response {#subject-search-response}
+The success response to a Subject Search Request is a Subject Search Response. It is a HTTP response of type `application/json`. Its body is a JSON object that contains the following fields:
+
+iat:
+: REQUIRED. The issued at time in `integer` format, expressed as epoch milliseconds
+
+subject:
+: REQUIRED. The subject for which the response is being issued. The format of this field is as described in the Subjects section ({{subjects}})
+
+results:
+: REQUIRED. An array of Subject Query results as described below ({{subject-query-result}}).
+
+reasons:
+: OPTIONAL. An array of Reason Objects ({{reason-object}}) that describe the reason for each reason identified in the `decisions` field. This field is REQUIRED if there is at least one decision in the `decisions` field that specifies a `reason_ids` field. The content of the `reasons` field MUST provide details of every identifier in the `reason_ids` fields in the `decisions` array.
+
+nextPageToken:
+: OPTIONAL. A string that MAY be used in a Search Request to fetch the next set of responses.
+
+#### Subject Query Result {#subject-query-result}
+A Subject Query Result is JSON object combining a subject, a list of resource attribute names and an action. Given that a Subject Query result is expected to be the response to a Subject Search, only positive matches should be returned; i.e., only those subjects that match the search criteria (those subjects that are allowed to access the provided Resource Attributes). Any Subjects absent from the results do not have any access to the Resource. 
+A Subject Query Result has the following fields:
```

Comment:
The example you write below is correct (except it should have the "decision" string in each object), but the description here needs to be updated to say that the response is an array of decisions. We should also call this top-level array something other than "decisions", since we are using that in the low-level decision key.

Reply:
See comment above for the 1st part: again, I think we should only return positive results, we're unnecessarily increasing the size of the response, and the computation required to create the response imho. Anything not returned should be assumed denied.

Reply:
Agreed with your previous comment. I feel we can drop the "reasons" from the Subject Search Query Response too.